### PR TITLE
Refactor Xassist, Test Framework

### PR DESCRIPTION
**Pull request by tharun571**
# Description

- Added xassist method to handle special cases.
- Updated OpenAI call method.
- Removed unnecessary line in testing framework.

## Type of change

Please tick all options which are relevant.

- [ ] Bug fix
- [ ] New feature
- [ ] Added/removed dependencies
- [ ] Required documentation updates

### CODE REVIEW

**Comment by anutosh491** [comment]
Just curious as to what exactly the refactor is doing here ? Maybe explain through a simple example ?

**Comment by tharun571** [comment]
> Just curious as to what exactly the refactor is doing here ? Maybe explain through a simple example ?

I was going through whatever code I wrote and checked for possible bugs. This PR takes care of scenarios when the prompt has special characters or anything involving disrupting the JSON to call the LLM.

File: src/xmagics/xassist.cpp [hunk 1]
```diff
@@ -279,6 +279,53 @@ namespace xcpp
...
+                    else
+                    {
+                        escaped += c;
+                    }
+                    break;
+            }
+        }
+        return escaped;
+    }
+
```
Comment (anutosh491) [inline src/xmagics/xassist.cpp:328]: > anything involving disrupting the JSON to call the LLM.

Hmm, so should there be a test required here as to check here to check this claim ?

Comment (tharun571) [reply src/xmagics/xassist.cpp:328]: > Hmm, so should there be a test required in this PR to check this claim ?

I did try but got in a predicament where `escape_special_cases("""{}") == "\"\"{}"` yields false but works in the kernel. So I tested them manually and pushed the code

**Review by vgvassilev — APPROVED** [review]
LGTM!

**Pull request merged**